pkg/virt-controller/watch/vm/vm.go

```diff
@@ -1678,10 +1678,10 @@ func getVMRevisionName(vmUID types.UID, generation int64) string {
 
 func patchVMRevision(vm *virtv1.VirtualMachine) ([]byte, error) {
 	vmCopy := vm.DeepCopy()
-	if revision.HasControllerRevisionRef(vmCopy.Status.InstancetypeRef) {
+	if vmCopy.Spec.Instancetype != nil && revision.HasControllerRevisionRef(vmCopy.Status.InstancetypeRef) {
```

Comment:
```
func HasControllerRevisionRef(ref *virtv1.InstancetypeStatusRef) bool {
	return ref != nil && ref.ControllerRevisionRef != nil && ref.ControllerRevisionRef.Name != ""
}
```
Isn't this handled already?

Reply:
That's checking `vm.status.{instancetypeRef,preferenceRef}` the issue here is that we didn't check that `vm.spec.{instancetype,preference}` wasn't `nil` before trying to set `vm.spec.{instancetype,preference}.revisionName`.